src/client/Components/Pages/Dashboard/ComponentStatusRow.jsx

```diff
@@ -0,0 +1,67 @@
+// @flow
+
+import React from 'react';
+
+import { makeGetRequest } from 'client/util';
+import ActionButton from 'client/Components/Pages/Dashboard/ActionButton';
+import { Link } from 'react-router-dom';
+
+type ComponentStatusRowProps = {
+  githubUsername: string,
+  githubRepo: string
+};
+
+type ComponentStatusRowState = {
+  isLoaded: boolean,
+  componentStatus: string
+};
+
+class ComponentStatusRow extends React.Component<ComponentStatusRowProps, ComponentStatusRowState> {
+  constructor(props: ComponentStatusRowProps) {
+    super(props);
+    this.state = {
+      isLoaded: false,
+      componentStatus: ''
+    };
+  }
+
+  componentDidMount(): void {
+    const url = `http://v9_website.ngrok.io/api/db/getComponentDashboardInfo?component=${this.props.githubRepo}`;
+    makeGetRequest(url).then((result) => {
+      let deploymentString = result.isDeploying ? 'Deploying' : '';
+      if (deploymentString === '') {
+        console.log('setting dep string');
+        deploymentString = result.deploymentIntention;
+      }
```

Comment:
This is a really weird way to write this

Reply:
You're a really weird way to write this